app/v1/controllers/CreateAndAmendCharitableGivingController.scala

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers
+
+import cats.data.EitherT
+import cats.implicits._
+import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.{Action, ControllerComponents}
+import utils.{IdGenerator, Logging}
+import v1.controllers.requestParsers.AmendCharitableGivingRequestReliefParser
+import v1.hateoas.HateoasFactory
+import v1.models.errors._
+import v1.models.request.createAndAmendCharitableGivingTaxRelief.CreateAndAmendCharitableGivingTaxReliefRawData
+import v1.models.response.createAndAmendCharitableGivingTaxRelief.CreateAndAmendCharitableGivingTaxReliefHateoasData
+import v1.models.response.createAndAmendCharitableGivingTaxRelief.CreateAndAmendCharitableGivingTaxReliefResponse.LinksFactory
+import v1.services.{AuditService, CreateAndAmendCharitableGivingTaxReliefService, EnrolmentsAuthService, MtdIdLookupService}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class CreateAndAmendCharitableGivingController @Inject() (val authService: EnrolmentsAuthService,
+                                                          val lookupService: MtdIdLookupService,
+                                                          parser: AmendCharitableGivingRequestReliefParser,
+                                                          service: CreateAndAmendCharitableGivingTaxReliefService,
+                                                          auditService: AuditService,
+                                                          hateoasFactory: HateoasFactory,
+                                                          cc: ControllerComponents,
+                                                          val idGenerator: IdGenerator)(implicit ec: ExecutionContext)
+    extends AuthorisedController(cc)
+    with BaseController
+    with Logging {
+
+  implicit val endpointLogContext: EndpointLogContext =
+    EndpointLogContext(controllerName = "AmendCharitableGivingController", endpointName = "createAndAmendCharitableGivingReliefs")
```

Comment:
controller name only mentions that it's an amend and not both a create and amend